spring-pulsar-spring-cloud-stream-binder/src/main/java/org/springframework/pulsar/spring/cloud/stream/binder/PulsarMessageChannelBinder.java

```diff
@@ -40,31 +54,76 @@ public class PulsarMessageChannelBinder extends
 		implements ExtendedPropertiesBinder<MessageChannel, PulsarConsumerProperties, PulsarProducerProperties> {
 
-	public PulsarMessageChannelBinder(PulsarTopicProvisioner provisioningProvider) {
+	private final PulsarTemplate<Object> pulsarTemplate;
+
+	private final PulsarConsumerFactory<?> pulsarConsumerFactory;
+
+	private PulsarExtendedBindingProperties extendedBindingProperties = new PulsarExtendedBindingProperties();
+
+	public PulsarMessageChannelBinder(PulsarTopicProvisioner provisioningProvider,
+			PulsarTemplate<Object> pulsarTemplate, PulsarConsumerFactory<?> pulsarConsumerFactory) {
 		super(null, provisioningProvider);
+		this.pulsarTemplate = pulsarTemplate;
+		this.pulsarConsumerFactory = pulsarConsumerFactory;
 	}
 
 	@Override
 	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
-			ExtendedProducerProperties<PulsarProducerProperties> producerProperties, MessageChannel errorChannel)
-			throws Exception {
-		return null;
+			ExtendedProducerProperties<PulsarProducerProperties> producerProperties, MessageChannel errorChannel) {
+
+		return message -> {
+			try {
+				PulsarMessageChannelBinder.this.pulsarTemplate.sendAsync(destination.getName(), message.getPayload());
+			}
+			catch (PulsarClientException e) {
+				// deal later
+			}
+		};
+
 	}
 
 	@Override
 	protected MessageProducer createConsumerEndpoint(ConsumerDestination destination, String group,
-			ExtendedConsumerProperties<PulsarConsumerProperties> properties) throws Exception {
-		return null;
+			ExtendedConsumerProperties<PulsarConsumerProperties> properties) {
+
+		PulsarContainerProperties pulsarContainerProperties = new PulsarContainerProperties();
+		pulsarContainerProperties.setTopics(new String[] { destination.getName() });
+		PulsarMessageDrivenChannelAdapter pulsarMessageDrivenChannelAdapter = new PulsarMessageDrivenChannelAdapter();
+		pulsarContainerProperties.setMessageListener((PulsarRecordMessageListener<?>) (consumer, msg) -> {
+			Message<Object> message = MessageBuilder.withPayload(msg.getValue()).build();
+			pulsarMessageDrivenChannelAdapter.send(message);
+		});
+		SchemaType schemaType = properties.getExtension().getSchemaType();
+		if (schemaType != null) {
+			pulsarContainerProperties.setSchema(toSchema(schemaType));
+		}
+		else {
+			pulsarContainerProperties.setSchema(Schema.BYTES);
+		}
+		pulsarContainerProperties.setSubscriptionName(properties.getExtension().getSubscriptionName());
+		DefaultPulsarMessageListenerContainer<?> container = new DefaultPulsarMessageListenerContainer<>(
+				this.pulsarConsumerFactory, pulsarContainerProperties);
+		pulsarMessageDrivenChannelAdapter.setMessageListenerContainer(container);
+		return pulsarMessageDrivenChannelAdapter;
+	}
+
+	// This is just a place-holder impl
+	public Schema<?> toSchema(SchemaType schemaType) {
```

Comment:
Got it - I see what you are doing here. I will add the API to the resolver in the next day or so.